File: backend/schemas/release_notes.py
"""Response schema for the public release-notes changelog.

A single per-version entry of the user-facing changelog served by
``GET /api/v1/release-notes`` (the *Aktualizácie* feature). Mirrors the
service-layer dict produced by
:func:`backend.services.release_notes.list_release_notes`.
"""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class ReleaseNote(BaseModel):
    """One version's user-facing release notes.

    Attributes:
        version: Version directory name, e.g. ``"v0.9.0"`` (drives both the
            card heading and the newest-first ordering).
        released_at: ISO date string (``YYYY-MM-DD``) of the release. Sourced
            from the ``versions`` table (``release_date``) when a matching row
            exists, otherwise the ``RELEASE_NOTES.md`` file mtime. ``None`` only
            in the degenerate case where neither is available.
        markdown: Raw Markdown body of the version's ``RELEASE_NOTES.md`` —
            rendered client-side.
    """

    version: str
    released_at: Optional[str] = None
    markdown: str

    model_config = {"from_attributes": True}

File: frontend/src/services/api/releaseNotes.ts
/**
 * API client for ``/api/v1/release-notes`` — the *Aktualizácie* changelog.
 *
 * Backend: :file:`backend/api/routes/release_notes.py` (public, no auth). Each
 * entry is one shipped version's user-facing release notes, newest first.
 *
 * The ``ReleaseNote`` shape is owned by ``nex-shared`` — it is the prop type of
 * the unified ``<ReleaseNotes>`` renderer (E1 unification) — and re-exported
 * here so this API client and the shared component can never drift apart.
 */

import { api } from "@/services/api";
import type { ReleaseNote } from "nex-shared";

export type { ReleaseNote };

/** List every shipped version's release notes, newest version first. */
export async function listReleaseNotes(): Promise<ReleaseNote[]> {
  return await api.get<ReleaseNote[]>("/release-notes");
}
